File: client/project-pokemon/src/app/models/put-pokemon-team-dto.ts
export interface PutPokemonTeamDto {
    nickname: string | null;
    shiny: boolean;
    sex: string | null;
    slot: number;
    teamId: number;
    pokemonId: number;
    natureId: number;
    movementId1: number;
    movementId2: number | null;
    movementId3: number | null;
    movementId4: number | null;
}

File: server/ProjectPokemon/ProjectPokemon/Models/Dtos/PokemonTeam/PutPokemonTeamDto.cs
namespace ProjectPokemon.Models.Dtos.PokemonTeam {
    public class PutPokemonTeamDto {
        public bool Shiny { get; set; } = false;
        public char? Sex { get; set; }
        public required int Slot { get; set; }
        public int NatureId { get; set; }
        public required int MovementId1 { get; set; }
        public int? MovementId2 { get; set; } = null;
        public int? MovementId3 { get; set; } = null;
        public int? MovementId4 { get; set; } = null;
    }
}